test/common/http/utility_test.cc

```diff
@@ -1495,12 +1495,18 @@ TEST(Url, ParsingFails) {
 }
 
 void validateUrl(absl::string_view raw_url, absl::string_view expected_scheme,
-                 absl::string_view expected_host_port, absl::string_view expected_path) {
+                 absl::string_view expected_host_port, absl::string_view expected_path,
+                 bool validate_tostring = true) {
   Utility::Url url;
   ASSERT_TRUE(url.initialize(raw_url, false)) << "Failed to initialize " << raw_url;
   EXPECT_EQ(url.scheme(), expected_scheme);
   EXPECT_EQ(url.hostAndPort(), expected_host_port);
   EXPECT_EQ(url.pathAndQueryParams(), expected_path);
+  if (validate_tostring) {
+    // Verify that url.toString() is equivalent to raw_url.
+    validateUrl(url.toString(), expected_scheme, expected_host_port, expected_path,
+                /*validate_tostring=*/false);
+  }
```

Comment:
This seems a bit hard to grok, seems like we're just testing that url.toString() does the right thing after parsing a string? I'm not sure if this is worth testing, but if there is a good reason I'd try to refactor this a bit to make it easier to understand what's going on

Reply:
Ping on this

Reply:
Reverted these changes, toString() is simple enough that it should be fine not to have dedicated tests for it.